File: Algorithms/Bubble_sort/bubble_sort.r
bubble_sort <- function(nums) {
  # swapped установлен в TRUE, чтобы цикл запустился хотя бы один раз
  swapped <- TRUE
  
  while (swapped) {
    swapped <- FALSE
    
    for (i in 1:(length(nums) - 1)) {
      if (nums[i] > nums[i + 1]) {
        # Элементы меняются местами
        temp <- nums[i]
        nums[i] <- nums[i + 1]
        nums[i + 1] <- temp
        
        # Устанавливаем swapped в TRUE для следующей итерации
        swapped <- TRUE
      }
    }
  }
  
  return(nums)
}

File: Algorithms/Bubble_sort/bubble_sort.jl
function bubble_sort!(arr)
    n = length(arr)
    for i in 1:n-1
        for j in 1:n-i
            if arr[j] > arr[j+1]
                arr[j], arr[j+1] = arr[j+1], arr[j]
            end
        end
    end
    return arr
end
